src/view/actions/updateVariable.jsx

```diff
@@ -327,34 +327,37 @@ const UpdateVariable = ({
   } = initInfo;
 
   useChanged(
-    useReportAsyncError(async () => {
-      setHasSchema(false);
-      setSelectedNodeId(null);
-
-      if (dataElement) {
-        const transforms = {};
-        const signal = abortPreviousRequestsAndCreateSignal();
-        const initialFormState = await getInitialFormStateFromDataElement({
-          dataElement,
-          context,
-          orgId,
-          imsAccess,
-          transforms,
-          existingFormStateNode: values,
-          signal
-        });
-
-        if (!signal.aborted) {
-          resetForm({ values: { ...initialFormState, dataElement } });
-          if (context.schema) {
-            setHasSchema(true);
+    useReportAsyncError(() => {
```

Comment:
React 18 doesn't like top-level async functions inside the useEffect hook. While this pattern worked in React 17, in React 18 the app breaks. I rewrote the function using the pattern recommended by the React team [here](https://react.dev/reference/react/useEffect#fetching-data-with-effects:~:text=You%20can%20also%20rewrite%20using%20the%20async%20/%20await%20syntax%2C%20but%20you%20still%20need%20to%20provide%20a%20cleanup%20function%3A).